File: test_doubles/csharp/TestDoubles/TestDoubles/Domain/PurchaseWorkflow.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace TestDoubles.Domain
{
    public class PurchaseWorkflow
    {
        private IDiscountLogic discountLogic;
        private Purchase purchase = new Purchase();

        public PurchaseWorkflow(IDiscountLogic discountLogic)
        {
            this.discountLogic = discountLogic;
        }

        public void AddItemsStep(params Item[] items)
        {
            foreach (Item item in items)
            {
                purchase.AddItem(item);
            }
        }

        public void DiscountStep()
        {
            discountLogic.ApplyDiscount(1254098L, "DISCOUNT_123", purchase);
        }
    }
}

File: test_doubles/java/TestDoubles/src/main/java/domain/PurchaseWorkflow.java
package domain;

import java.util.Arrays;

public class PurchaseWorkflow {
    private DiscountLogic discountLogic;
    private Purchase purchase = new Purchase();

    public PurchaseWorkflow(DiscountLogic discountLogic) {
        this.discountLogic = discountLogic;
    }

    public void addItemsStep(Item... items) {
        Arrays.stream(items).forEach(purchase::addItem);
    }

    public void discountStep() {
        discountLogic.applyDiscount(1254098L, "DISCOUNT_123", purchase);
    }
}
